### PR TITLE
Refactor db access

### PR DESCRIPTION
**Pull request by michibusch**
On a high level, this branch:
- deprecates all write access to DerbyDB
- is able to read from Derby and will migrate all AudioItem and Playlist data into a Lucene index on startup (which means the first ACM startup takes longer due to the migration step)
- will store the Lucene index in Dropbox and remove the old DB files from Dropbox on the first Dropbox commit; after that it will only read/write from the Lucene index
- significantly simplifies all the classes handling AudioItem metadata; it removes (or deprecates) a lot of redundant classes, e.g. AudioItem vs. PersistentAudioItem; 
- gets rid of the annoying "unable to connect to database" bug on windows
- improves startup and general performance

### CODE REVIEW

File: acm/src/main/java/org/literacybridge/acm/config/DBConfiguration.java [hunk 1]
```diff
@@ -165,22 +153,33 @@ public void init() {
             controlAccess = new ControlAccess(this);
             controlAccess.init();
 
+            final Taxonomy taxonomy = Taxonomy.createTaxonomy(sharedACMDirectory);
+            if (!AudioItemIndex.indexExists(getLuceneIndexDirectory())) {
```
Comment (billev) [inline acm/src/main/java/org/literacybridge/acm/config/DBConfiguration.java]: What about the case of a brand new ACM? Wouldn't that case not have a Lucene Index? In which case there would be nothing to migrate.

Comment (michibusch) [inline acm/src/main/java/org/literacybridge/acm/config/DBConfiguration.java]: Good catch! Fixed in the latest diff.

File: acm/src/main/java/org/literacybridge/acm/api/IDataRequestResult.java [hunk 2]
```diff
@@ -12,30 +10,18 @@
...
  */
 public interface IDataRequestResult {
-	/**
-	 * Returns the root category which should be used to walk
-	 * the tree of categories. The root category is a virtual root
-	 * and should generally not be displayed by the UI.
-	 */
-	public abstract Category getRootCategory();
+    /**
+     * Returns the facet count for the passed-in category.
```
Comment (billev) [inline acm/src/main/java/org/literacybridge/acm/api/IDataRequestResult.java]: "Facet" seems to be a key Lucene concept, but I can't find any definition or description of how that relates to or is used by the ACM.

File: acm/src/main/java/org/literacybridge/acm/api/IDataRequestResult.java [hunk 3]
```diff
@@ -2,9 +2,7 @@
 
 import java.util.List;
 
-import org.literacybridge.acm.categories.Taxonomy.Category;
-import org.literacybridge.acm.content.AudioItem;
-import org.literacybridge.acm.db.PersistentTag;
+import org.literacybridge.acm.store.Category;
 
 /**
  * This is the result object returned by {@link IDataRequestService} containing all necessary
```
Comment (billev) [inline acm/src/main/java/org/literacybridge/acm/api/IDataRequestResult.java]: Please explain how this contains "all necessary data to render the UI".

File: acm/src/main/java/org/literacybridge/acm/core/DataRequestResult.java [hunk 4]
```diff
@@ -4,70 +4,52 @@
 import java.util.Map;
 
 import org.literacybridge.acm.api.IDataRequestResult;
-import org.literacybridge.acm.categories.Taxonomy.Category;
-import org.literacybridge.acm.db.PersistentTag;
+import org.literacybridge.acm.store.Category;
 
 public class DataRequestResult implements IDataRequestResult {
```
Comment (billev) [inline acm/src/main/java/org/literacybridge/acm/core/DataRequestResult.java]: As the concrete implementation of IDataRequestResult, this should probably document just what it (concretely) is.

File: acm/src/main/java/org/literacybridge/acm/core/DataRequestService.java [hunk 5]
```diff
@@ -1,114 +1,55 @@
...
-import org.literacybridge.acm.db.PersistentCategory;
-import org.literacybridge.acm.db.PersistentLocale;
-import org.literacybridge.acm.db.PersistentTag;
-import org.literacybridge.acm.gui.Application;
-import org.literacybridge.acm.index.AudioItemIndex;
+import org.literacybridge.acm.store.Category;
+import org.literacybridge.acm.store.MetadataStore;
+import org.literacybridge.acm.store.Playlist;
 
 public class DataRequestService implements IDataRequestService {
```
Comment (billev) [inline acm/src/main/java/org/literacybridge/acm/core/DataRequestService.java]: As with DataRequestResult, this class should probably describe what it (concretely) is.

File: acm/src/main/java/org/literacybridge/acm/api/IDataRequestResult.java (hunk 2 above)
Comment (michibusch) [inline acm/src/main/java/org/literacybridge/acm/api/IDataRequestResult.java]: In my latest commit I removed the old interface IDataRequestService and class DataRequestService and changed the places that used it to use the new MetadataStore directly. Also, I removed IDataRequestResult and renamed DataRequestResult to SearchResult. I added documentation to SearchResult describing what 'Facet count' means (it's actually not only a Lucene concept)

**Pull request merged**